inc/Utf8/Sort.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace dokuwiki\Utf8;
+
+/**
+ * DokuWiki sort functions
+ *
+ * When "intl" extension is available, all sorts are done using a collator.
+ * Otherwise, primitive PHP functions are called.
+ *
+ * The collator is created using the locale given in $conf['lang'].
+ * It always uses case insensitive "natural" ordering in its collation.
+ * The fallback solution uses the primitive PHP functions that return almost the same results
+ * when the input is text with only [A-Za-z0-9] characters.
+ *
+ * @license    GPL 2 (http://www.gnu.org/licenses/gpl.html)
+ * @author     Moisés Braga Ribeiro <moisesbr@gmail.com>
+ */
+class Sort
+{
+    /** @var \Collator[] language specific collators, usually only one */
+    protected static $collator;
```

Comment:
Clever trick. An array makes function lang_has_changed() not necessary.
But, as it is now an array, the variable name should be $collators (plural).